BepInEx.Core/Paths.cs

```diff
@@ -104,20 +104,31 @@ public static void SetExecutablePath(string executablePath,
 
         GameRootPath = PlatformHelper.Is(Platform.MacOS)
                            ? Utility.ParentDirectory(executablePath, 4)
-                           : Path.GetDirectoryName(executablePath);
+                           : Path.GetDirectoryName(executablePath) 
+                          ?? throw new DirectoryNotFoundException("Failed to extract GameRootPath from executablePath: " + executablePath);
+
+        if (managedPath != null && gameDataRelativeToManaged)
```

Comment:
How does `gameDataRelativeToManaged` work? When would it ever be the case?

Reply:
It's true in mono net35.